src/main/java/Parser.java

```diff
@@ -0,0 +1,152 @@
+import CustomExceptions.*;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class Parser {
+    private final String input;
+    private static final ArrayList<String> validActions
+            = new ArrayList<>(Arrays.asList("todo", "deadline", "event", "done", "delete", "list", "bye"));
+
+    public Parser(String input) {
+        this.input = input;
+    }
+
+    public String getAction() {
+        return (input + " ").split(" ")[0].toLowerCase();
+    }
+
+    private String getRemainingTokens() {
+        if (!input.contains(" ")) {
+            return "";
+        }
+        return this.input.split(" ", 2)[1];
+    }
+
+    public String getDescription() {
+        String action = this.getAction();
+        String remainingTokens = this.getRemainingTokens();
+
+        if (action.equals("deadline") && remainingTokens.contains("/by")) {
+            return remainingTokens.split("/by")[0].trim();
+        } else if (action.equals("event") && remainingTokens.contains("/at")) {
+            return remainingTokens.split("/at")[0].trim();
+        } else {
+            return remainingTokens;
+        }
+    }
+
+    public LocalDateTime getBy() {
+        return convertToDateTime(this.getByString());
+    }
+
+    public LocalDateTime getAt() {
+        return convertToDateTime(this.getAtString());
+    }
+
+    private String getByString() {
+        String action = this.getAction();
+        String remainingTokens = this.getRemainingTokens();
+        if (action.equals("deadline") && remainingTokens.contains("/by")) {
+            return remainingTokens.split("/by", 2)[1].trim();
+        }
+        return "";
+    }
+
+    private String getAtString() {
+        String action = this.getAction();
+        String remainingTokens = this.getRemainingTokens();
+        if (action.equals("event") && remainingTokens.contains("/at")) {
+            return remainingTokens.split("/at", 2)[1].trim();
+        }
+        return "";
+    }
+
+    public boolean inputsAreValid() {
```

Comment:
Instead of naming inputsAreValid, areInputsValid seem more inline for a boolean return value method.